softlayer/resource_softlayer_file_storage.go

```diff
@@ -594,18 +594,20 @@ func WaitForStorageAvailable(d *schema.ResourceData, meta interface{}) (interfac
 
 			// Check volume status.
 			log.Println("Checking volume status.")
-			resultStr := ""
+			var resultObj interface{}
 			err = sess.DoRequest(
 				"SoftLayer_Network_Storage",
 				"getObject",
 				nil,
 				&sl.Options{Id: &id, Mask: "volumeStatus"},
-				&resultStr,
+				&resultObj,
 			)
```

Comment:
Please refer to the following link : http://stackoverflow.com/questions/41906025/softlayer-network-storage-volumestatus-is-not-a-string-type